plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -1,14 +1,49 @@
-@objc(initialize:)
-  func initialize(command: CDVInvokedUrlCommand) {
-    let args = command.arguments
-    
-    let inAppConsentStrategy: InAppConsentStrategy = {
-      if let strategy = args?[2] as? String {
-        return .init(rawValue: strategy)!
-      }
-      return.notEnabled
-    }()
-
-    let config = OptimoveConfigBuilder(optimoveCredentials: args?[0] as? String, optimoveCredentials: args?[1] as? String).enableInAppMessaging(inAppConsentStrategy: inAppConsentStrategy).build()
-    Optimove.initialize(with: config)
-  }
\ No newline at end of file
+import OptimoveSDK
+import NotificationCenter
+@objc(Optimove_Cordova) class OptimobileSDKPlugin : CDVPlugin {
+    private let internalQueue = DispatchQueue(label: "com.singletioninternal.queue",
+                                              qos: .default,
+                                              attributes: .concurrent)
+    
+    private var pendingPush: PushNotification?
+    
+    private let optimoveCredentialsKey = "optimoveCredentials"
+    private let optimoveMobileCredentialsKey = "optimoveMobileCredentials"
+    
+    func load() {
+        internalQueue.sync(flags: .barrier) {
+            NotificationCenter.default.addObserver(self, selector: #selector(self.didFinishLaunching(notification:)), name: UIApplication.didFinishLaunchingNotification, object: nil)
+        }
+    }
+    
+    @objc func didFinishLaunching(notification: NSNotification) {
+        let configPath = Bundle.main.path(forResource: "optimove", ofType: "plist")
+        
+        guard let configPath = configPath else {
+            print("optimove.plist NOT FOUND");
+            return
+        }
+        
+        guard let configValues: [String: String] = NSDictionary(contentsOfFile: configPath) as? [String: String] else {
+            print("optimove.plist IS NOT VALID");return
+        }
+        
+//        if let userInfo = notification.userInfo {
+//            let userInfoDict = notification.userInfo?[UIApplication.LaunchOptionsKey.remoteNotification];
+//            if userInfoDict != nil {
+//                pendingPush = PushNotification.init(userInfo: userInfo, response: nil)
+//            }
+//        }
+        
+//        let config = OptimoveConfigBuilder(optimoveCredentials: configValues[optimoveCredentialsKey], optimobileCredentials: configValues[optimoveMobileCredentialsKey])
```

Comment:
We should still have the setup of config from the plist values in here?

Reply:
you're right:)

Reply:
done